docs/examples/go-experimental/getting-started.md

```diff
@@ -0,0 +1,118 @@
+Go doesn’t support decorators, so indicating which classes should be deployed is done using the `genezio.yaml` configuration file (see [Configuration File - Classes](../../project-structure/genezio-configuration-file#classes)). An yaml configuration for the project structure mentioned above would look like this:
+
+```yaml
+name: test-go
+region: eu-central-1
+classes:
+  - path: ./users/users.go
+    type: jsonrpc
+  - path: ./tasks/tasks.go
+    type: jsonrpc
+```
+
+### “Classes”
+
+GO doesn’t have the notion of “classes”. We have, thankfully, an alternative. Structs with attached methods. For the rest of this page, we will refer to them as `genezio classes`, or simply `classes`. Our parser will choose and analyze the first struct it finds (not necessarily the first declared) that has the following properties:
+
+- is exported (starts with a capital letter)
+- has at least one method attached
+- the package exports a method named `New` which takes no parameters and returns an instance of the struct
+
+### Methods
+
+Each method that will eventually be deployed needs to respect some signature rules:
+
+- a method can take any number of parameters as long as they are serializable (see [Serialization rules](#serialization-rules))
+- it returns an error or a tuple formed by a type and an error (error can be either the standard interface defined in the standard library, or any custom type implementing that interface)
+
+```go
+func (u UserService) Get(id int) (*User, error) // valid
+func (u *UserService) Delete(user User) error // valid
+func (u *UserService) GetAge(user *User) int // NOT VALID, shlould return (int, error)
+```
+
+Notice that we can attach methods to the value of a struct as well as to a pointer of that struct. You can use and mix those kinds of declarations as you wish. It makes no difference for the SDK generation or the way your code runs on the genezio infrastructure.
+
+Example of a valid “genezio class” in GO:
+
+```go
+package users
+
+import (
+	"fmt"
+
+	"go-example/models"
+)
+
+// the genezio class
+type UserService struct{}
+
+func New() UserService {
+	return UserService{}
+}
+
+func (u UserService) GetAge(name string, age *int) (int, error) {
+	fmt.Println("Get Age for", name)
+	if age == nil {
+		return 0, fmt.Errorf("age is nil")
+	}
+	return *age, nil
+}
+
+func (u *UserService) Unimplemented() error {
+	fmt.Println("Unimplemented")
+	return fmt.Errorf("not implemented")
+}
+
+func (u *UserService) Delete(user *models.User) (*models.User, error) {
+	if user == nil {
+		return nil, fmt.Errorf("user is nil")
+	}
+	fmt.Println("Deleted user")
+	return user, nil
+}
+```
+
+Notice you can import types from other packages. They also have to satisfy the same rules for serialization.
+
+### Serialization rules
+
+- primitive types (int, float64, bool, string), are serializable
+- structs are serializable
+- arrays are serializable
+- maps are serializable as long as the key is `string` and the value type is serializable
```

Comment:
I will guess this limitation comes from the fact that you represent TS maps as:
```ts
let map: {[key: string]: string};
```

What if we change this and use the [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) type in JavaScript?
```ts
let map: Map<string, string>;
```

This would allow us to remove this annoying limitation that has nothing to do with Golang.

Reply:
this limitation comes from the fact that we use json to send data over the wire, any other key type other than `string` would result in serialization errors